pkg/sources/reconciler/awssnssource/subscription.go

```diff
@@ -228,6 +252,23 @@ func findSubscription(ctx context.Context, cli snsiface.SNSAPI, topicARN, endpoi
 	return "", awserr.New(sns.ErrCodeNotFoundException, "", nil)
 }
 
+// checkTopicExists returns whether the topic with the given ARN exists.
+func checkTopicExists(ctx context.Context, cli snsiface.SNSAPI, topicARN string) (bool, error) {
+	in := &sns.GetTopicAttributesInput{
+		TopicArn: &topicARN,
+	}
+
+	_, err := cli.GetTopicAttributesWithContext(ctx, in)
```

Comment:
Note to self: implement this method in the mocked SNS client. Tests are currently failing:

```
=== RUN   TestReconcileSubscription/Deletion_while_subscribed
    --- FAIL: TestReconcileSubscription/Deletion_while_subscribed (0.03s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
...
```